platforms/web/quality-assurance.md

```diff
@@ -0,0 +1,157 @@
+- Third party API integrations that can & will be re-used across projects should be implemented as external packages, assuming no existing open-source integration exists, to allow for independent testing and code reuse if appropriate.
+- Ensure third party API keys (e.g. SES keys) are not present within .env.example and other relevant files outside of the production / development environments.
+
+### Controllers
+
+- Be on the lookout for duplicated code - recommend refactoring into helper classes or models if so
+- Ensure all requests are validated using a custom request
+- Errors are handled correctly in try catch blocks
+- Responses are returned in a consistent & logical format
+- Only appropriate & relevant information is returned
+- When returning JSON ensure the returned JSON conforms to the JSON spec (http://json-schema.org/specification.html)
+- Appropriate status codes are set
+
+### Emails
+
+- Emails use the SES driver unless agreed beforehand
+- All emails are queued up for sending
+- Only relevant information is passed to the mailable instance
+
+### Views
+
+- Loading of assets (js/html) use the `asset` global helper
+- Blade layout system is used where appropriate (@extends, @yield)
+- Components are split out logically where appropriate, make use of Laravel’s powerful blade templating system if appropriate (clientCard, navbar etc)
+
+### Service Providers
+
+- Service Providers are utilised where appropriate to bind classes into the container (API interfaces etc)
+
+### File Storage
+
+- Files are stored on S3 unless requested by the client to be stored locally or agreed otherwise beforehand by the project manager
+- Ensure appropriate permissions on files in applicable (sensitive documents are private, avatars are public etc)
+
+### Queues
+
+- Queues use Redis unless agreed beforehand by the project manager
+- Ensure appropriate measures are in place for handling failed jobs
+
+### Sockets
+
+- When using PHP, sockets are integrated using Laravel Echo unless agreed beforehand by the project manager
+- Ensure appropriate authentication & security for each connection
+
+### Database
+
+- All migrations use foreign keys for relations to ensure database integrity
+- SQL data types are appropriate
+  - Monetary values should either be stored in pence as an INTEGER or as a DECIMAL - Never a FLOAT
+  - Long strings (descriptions) should be stored as TEXT, other string can be stored as a VARCHAR - Ensure controller validation for VARCHAR fields have a max string length of 255
+  - Tables are named appropriately with underscores in lieu of other word separators
+  
+  
+### Cache
+
+- Ensure appropriate values are cached where appropriate.
+
+### Continuous Integration
+
+- Is used, even if it's just a basic setup
+- If Travis CI is being used, ensure third party API keys are input directly into Travis CI’s website relevant to the project and make sure “Display value in build log” is not checked.
+
+
+
+
```

Comment:
There's a lot of empty lines here

Reply:
This should be fixed in the latest commit 👍